sprockets/mixins/mediatype/content.py

```diff
@@ -86,17 +95,24 @@ def make_application():
     instead.
 
     """
-    def __init__(self):
+
+    default_content_type: typing.Union[str, None]
```

Comment:
Why are some of these nullable fields `Union[T, None]` and others `Optional[T]`?

Reply:
Ya, much cleaner to go with `typing.Optional`

Reply:
I use `typing.Optional[T]` whenever it is an optional parameter and `typing.Union[T, None]` for things that are either of type `T` or `None`.

Personally I wish that [typing.Optional](https://docs.python.org/3/library/typing.html#typing.Optional) was never introduced because a name that is bound to either an instance of a type `T` or `None` is not at all optional IMO.  Keyword parameters in `**kwargs` or a positionals in `*args` are good examples of things that are optional.